tests/integration/suite/actors/healthz/deactivate-on-unhealthy.go

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implieh.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package healthz
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strconv"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	chi "github.com/go-chi/chi/v5"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/slices"
+
+	"github.com/dapr/dapr/tests/integration/framework"
+	"github.com/dapr/dapr/tests/integration/framework/process/daprd"
+	prochttp "github.com/dapr/dapr/tests/integration/framework/process/http"
+	"github.com/dapr/dapr/tests/integration/framework/process/placement"
+	"github.com/dapr/dapr/tests/integration/framework/util"
+	"github.com/dapr/dapr/tests/integration/suite"
+)
+
+func init() {
+	suite.Register(new(deactivateOnUnhealthy))
+}
+
+// deactivateOnUnhealthy ensures that all active actors are deactivated if the app is reported as unhealthy.
+type deactivateOnUnhealthy struct {
+	daprd               *daprd.Daprd
+	place               *placement.Placement
+	isHealthy           atomic.Bool
```

Comment:
You are right. This was left over from when I was trying to do the test with making the app unhealthy - but that required waiting A LONG TIME for the healthchecks to kick in. I'll remove it.

Reply:
We still need to rename the test/comments etc to be about disconnecting from placement, rather than healthz of the app.

I think it is also required for us to have a test to ensure that deactivation occurs when the app is made unhealthy. I think we should evaluate the [current settings](https://github.com/dapr/dapr/blob/90835fafb97314b33786134b78241e8e45333328/pkg/actors/actors.go#L285) for actor health checking which I don't think are very appropriate. Along with https://github.com/dapr/dapr/pull/7022, I think we should unify the check behaviour with the normal app health check- why are they not the same anyway? That work should be a dependency of this change imo.

Reply:
I'll update the comments.

When #7022 is merged we can also make a test for the app becoming un-healthy.

As for unifying actor health checks and app health checks if enabled... Agreed: #6216

Reply:
Done with the renaming